crates/ruma-common/src/directory.rs

```diff
@@ -223,6 +228,29 @@ impl<'de> Visitor<'de> for RoomNetworkVisitor {
     }
 }
 
+/// The rule used for users wishing to join a public room.
+///
+/// This type can hold an arbitrary string. To check for formats that are not available as a
+/// documented variant here, use its string representation, obtained through `.as_str()`.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+#[serde(tag = "join_rule")]
```

Comment:
```suggestion
#[serde(tag = "join_rule", rename_all = "lowercase")]
```

(or `snake_case`, maybe?)